backend/question-service/lambda/app/scraper/handler.py

```diff
@@ -0,0 +1,189 @@
+    
+    return None # If the script tag is not found, return None
+
+def scrape_leetcode_from_urls(urls_to_scrape):
+    """Scrape Leetcode for the questions in the list of URLs concurrently and return a list of question details."""
+    problems = []
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = [executor.submit(scrape_leetcode_from_url, url) for url in urls_to_scrape]
+        for future in concurrent.futures.as_completed(futures):
+            result = future.result()
+            if result:
+                problems.append(result)
+
+    return problems
+
+def get_details_from_queries(queries: list[dict]):
+    """Return the question details from the question's data queries."""
+    question_details = {}
+
+    # Set updated_at timestamp
+    question_details["updated_at"] = int(time.time())
+
+    # Extract question details from the first query
+    data = queries[0].get("state", {}).get("data", {}).get("question", {})
+    question_details["title"] = data.get("title", "")
+    question_details["url"] = f"https://leetcode.com/problems/{data.get('titleSlug', '')}/"
+    question_details["difficulty"] = data.get("difficulty", "")
+    question_details["id"] = data.get("questionId", "")
+    question_details["paid_only"] = data.get("isPaidOnly", "")
+    question_details["category"] = data.get("categoryTitle", "")
+
+    # If question is paid only, return the question details without making any more queries as no additional information is available
+    if question_details["paid_only"]:
+        return question_details
+
+    # Extract language list from the second query
+    languages = queries[1].get("state", {}).get("data", {}).get("languageList", [])
+    question_details["languages"] = [lang["name"] for lang in languages]
+
+    # Extract example test cases from the fourth query
+    test_cases = queries[3].get("state", {}).get("data", {}).get("question", {}).get("exampleTestcaseList", [])
+    question_details["test_cases"] = test_cases
+
+    # Extract content from the seventh query
+    question_details["description"] = BeautifulSoup(queries[6].get("state", {}).get("data", {}).get("question", {}).get("content", ""), 'html.parser').text
+
+    # Extract hints from the sixth query
+    hints = queries[5].get("state", {}).get("data", {}).get("question", {}).get("hints", [])
+    question_details["hints"] = [BeautifulSoup(hint, 'html.parser').text for hint in hints]
+
+    # Extract topic tags from the fifth query
+    tags = queries[8].get("state", {}).get("data", {}).get("question", {}).get("topicTags", [])
+    question_details["topic_tags"] = [tag["name"] for tag in tags]
+
+    # Extract code snippets from the last query
+    code_snippets = queries[-1].get("state", {}).get("data", {}).get("question", {}).get("codeSnippets", [])
+    question_details["code_snippets"] = code_snippets
+
+    # Return extracted data
+    return question_details
+
+def connect_to_mongo(db_url, retries=5, delay=5):
```

Comment:
nit: will abstracting the helper functions to another file significantly complicate the process to deploy to Lambda? if no I think its a nice to have for better readability & separation of concerns